daemon/daemon_test.go

```diff
@@ -341,6 +341,106 @@ func TestDaemonReloadLabels(t *testing.T) {
 	}
 }
 
+func TestDaemonReloadMirrors(t *testing.T) {
```

Comment:
I wonder if we should simplify this test, and have a single slice of test-cases, each with "before", "new", and "expected", and a property that indicates if it should fail.